libs/entitlement/src/test/java/org/elasticsearch/entitlement/runtime/policy/PolicyParserTests.java

```diff
@@ -39,58 +35,44 @@ public void testGetEntitlementTypeName() {
     public void testPolicyBuilder() throws IOException {
         Policy parsedPolicy = new PolicyParser(PolicyParserTests.class.getResourceAsStream("test-policy.yaml"), "test-policy.yaml", false)
             .parsePolicy();
-        Policy builtPolicy = new Policy(
+        Policy expected = new Policy(
             "test-policy.yaml",
             List.of(new Scope("entitlement-module-name", List.of(new FileEntitlement("test/path/to/file", List.of("read", "write")))))
         );
-        assertEquals(parsedPolicy, builtPolicy);
+        assertEquals(expected, parsedPolicy);
     }
 
     public void testPolicyBuilderOnExternalPlugin() throws IOException {
         Policy parsedPolicy = new PolicyParser(PolicyParserTests.class.getResourceAsStream("test-policy.yaml"), "test-policy.yaml", true)
             .parsePolicy();
-        Policy builtPolicy = new Policy(
+        Policy expected = new Policy(
             "test-policy.yaml",
             List.of(new Scope("entitlement-module-name", List.of(new FileEntitlement("test/path/to/file", List.of("read", "write")))))
         );
-        assertEquals(parsedPolicy, builtPolicy);
+        assertEquals(expected, parsedPolicy);
     }
 
     public void testParseCreateClassloader() throws IOException {
         Policy parsedPolicy = new PolicyParser(new ByteArrayInputStream("""
             entitlement-module-name:
               - create_class_loader
             """.getBytes(StandardCharsets.UTF_8)), "test-policy.yaml", false).parsePolicy();
-        Policy builtPolicy = new Policy(
+        Policy expected = new Policy(
             "test-policy.yaml",
             List.of(new Scope("entitlement-module-name", List.of(new CreateClassLoaderEntitlement())))
         );
-        assertThat(
```

Comment:
We previously needed these complex assertions because the non-record entitlement classes didn't define `equals` and `hashCode`. Now that they're well behaved, `assertEquals` works.